### PR TITLE
Added kornia resize in object detection

### LINKED ISSUE #2439: Use Kornia resize in ResizePreprocessor in ObjectDetector
Use Kornia resize in ResizePreprocessor

_Originally posted by @edgarriba in https://github.com/kornia/kornia/pull/2363#discussion_r1257304346_

### PR DESCRIPTION
**Pull request by jeffin07**
#### Changes
Fixes #2439 

Added kornia resize in Object detection

Fixes # (issue)

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/contrib/object_detection.py [hunk 1]
```diff
@@ -22,13 +22,14 @@ def __init__(self, size: int | tuple[int, int], interpolation_mode: str = "bilin
         """
         super().__init__()
         self.size = size
+        self.resizer = Resize((self.size, self.size))
```
Comment (edgarriba) [inline kornia/contrib/object_detection.py:25]:
```suggestion
        self.resizer = Resize(self.size)
```

File: kornia/contrib/object_detection.py [hunk 2]
```diff
@@ -3,9 +3,9 @@
 from typing import Any
 
 import torch
-import torch.nn.functional as F
 
 from kornia.core import Module, Tensor, concatenate
+from kornia.geometry import Resize
```
[the remark this answers is not in the document]
Comment (edgarriba) [reply kornia/contrib/object_detection.py:8]: for internal development we tend to import with path as deep as possible

File: kornia/contrib/object_detection.py (hunk 1 above)
Comment (jeffin07) [reply kornia/contrib/object_detection.py:25]: @edgarriba  If size is an int, then smaller edge of the image will be matched to size right?

Comment (johnnv1) [reply kornia/contrib/object_detection.py:25]: yes

[the remark this answers is not in the document]
Comment (johnnv1) [reply kornia/contrib/object_detection.py:25]: and i think we can remove the `self.interpolation_mode = interpolation_mode` because it's just used for the old resize

Comment (edgarriba) [reply kornia/contrib/object_detection.py:25]: i believe that this logic is already handled by the Resize class

Comment (jeffin07) [reply kornia/contrib/object_detection.py:25]: @johnnv1 I tried using `side` parameter but only one  dimension of the image is resized. I tried all the possible values for `side` parameter

Comment (jeffin07) [reply kornia/contrib/object_detection.py:25]: These were the test-cases i tried for `side` parameter. And only one dimension is getting resized when `size` is an integer

```bash

>>> from kornia.geometry.transform import Resize
>>> res = Resize(400, side="long", interpolation="bilinear")
>>> img = torch.randn(3, 600, 300, device="cpu", dtype=torch.float32)
>>> img_resized = res(img)
>>> img_resized.shape 
torch.Size([3, 400, 200])

//  side="vert"

>>> res = Resize(400, side="vert", interpolation="bilinear")
>>> img_resized.shape
torch.Size([3, 400, 200])

// side="horz"

>>> res = Resize(400, side="horz", interpolation="bilinear")
>>> img_resized = res(img)
>>> img_resized.shape
torch.Size([3, 800, 400])

// side="short"

>>> res = Resize(400, side="short", interpolation="bilinear")
>>> img_resized = res(img)
>>> img_resized.shape
torch.Size([3, 800, 400])



```

Comment (johnnv1) [reply kornia/contrib/object_detection.py:25]: These examples look right to me 

- when `vert` the vertical side (height -- dim 1) is resized to 400, and the width keeps the aspect ratio and goes to 200
- when `short` the shortest side (width on your ex image) is resized to 400, and the height goes to 800 to keep the aspect ratio

if you want to resize both sizes for the exact size you need to resize then pad the image (not causing any distortion keeping the aspect ratio)

Comment (jeffin07) [reply kornia/contrib/object_detection.py:25]: @johnnv1 So If i gave  an interger for `size` in `Resize`, Is there any parameter for `Resize` class  to resize both the dimention of the image

> i believe that this logic is already handled by the Resize class

Comment (johnnv1) [reply kornia/contrib/object_detection.py:25]: i belive not, this case we should pass a tuple

Comment (jeffin07) [reply kornia/contrib/object_detection.py]: @johnnv1 Updated the changes

File: kornia/contrib/object_detection.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeffin07) [reply kornia/contrib/object_detection.py]: Done

**Review by edgarriba — APPROVED** [review]

**Review by johnnv1 — APPROVED** [review]
